nsqd/nsqd.go

```diff
@@ -550,7 +572,16 @@ func (n *NSQD) DeleteExistingTopic(topicName string) error {
 	return nil
 }
 
-func (n *NSQD) Notify(v interface{}) {
+func (n *NSQD) Notify(v interface{}, notifyType int) {
+	var nc chan interface{}
+	switch notifyType {
+	case NotifyTypeRegistration, NotifyTypeUnRegistration:
+		nc = n.notifyChan
+	case NotifyTypeStateUpdate:
+		nc = n.stateNotifyChan
+	default:
+		nc = n.notifyChan
+	}
```

Comment:
Seems like it might come out simpler if we use the same channel, but send a more complex type across the channel to perform logic on the channel receive side.